### PR TITLE
Refactor of ground truth formatting and merging 

### PR DESCRIPTION
**Pull request by CasperKjaerhus**
This refactor was made because of a new ground truth data that is in drive right now.

* We only keep the speed limit from vejman, since only that should be used for evaluating models
    * however its possible to add the other features by adding their coloumns to line 50 in GroundTruthHandler

### CODE REVIEW

File: pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import os
+
+
+class GroundTruthHandler:
+    """This class handles formatting the "dirty" ground truth dataset and
+    also handles merging the ground truth with the FCD dataset for use in model training
+    """
+
+    def __init__(self):
+        pass
```
Comment (AlexanderHHansen) [inline pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py:13]: bare fjern init, hvis du bare pass'er

File: pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    @classmethod
+    def load_from_geojson(cls, file: TextIOWrapper, year: int):
+        """Loads the ground truth from the dirty geojson and cleans it
+
+        Args:
+            file (TextIOWrapper): the file of the ground truth geojson.
+            year (int): The year that the ground truth speed limit is from at the latest.
+
+        Returns:
+            _type_: A dataframe that has 2 coloumns: osm_id and KODE_HAST_GENEREL_HAST, the latter being the speed limit
```
Comment (AlexanderHHansen) [inline pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py:24]: missing type

File: pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        """
+        data = json.load(file)
+        return cls.__clean_vejman_data(data, year)
+
+    @classmethod
+    def __clean_vejman_data(cls, df: pd.DataFrame, year: int) -> pd.DataFrame:
+        """Cleans the dataframe by removing the superflous coloumns along
+            with filtering speed limits that have been changed too recently.
```
Comment (AlexanderHHansen) [inline pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py:32]: Cleans the dataframe by removing the unnecessary columns along
            with filtering, speed limits that have been changed too recently.

File: pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+        # Type both osm_id and speed limit as int
+        cleandf = cleandf.astype(int)
+
+        return cleandf
+
+    @classmethod
+    def add_ground_truth_to_FCD_Data(
+        cls, fcd_data: pd.DataFrame, ground_truth_data: pd.DataFrame
+    ):
```
Comment (AlexanderHHansen) [inline pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py:61]: missing return type

File: pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        """Adds the ground truth speed limit as a coloumn KODE_HAST_GENEREL_HAST to a dataframe by matching the osm_id coloumn.
+
+        Args:
+            fcd_data (pd.DataFrame): FCD data with osm_id coloumn
+            ground_truth_data (pd.DataFrame): Clean ground truth speed limit data with osm_id and KODE_HAST_GENEREL_HAST coloumns
+
+        Returns:
+            _type_: Dataframe which has the same coloumns as fcd_data input but with KODE_HAST_GENEREL_HAST coloumn added.
```
Comment (AlexanderHHansen) [inline pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py:69]: missing type

File: pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+class GroundTruthHandler:
+    """This class handles formatting the "dirty" ground truth dataset and
+    also handles merging the ground truth with the FCD dataset for use in model training
+    """
+
+    def __init__(self):
+        pass
+
+    @classmethod
+    def load_from_geojson(cls, file: TextIOWrapper, year: int):
```
Comment (AlexanderHHansen) [inline pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py:16]: missing return type

**Review by AlexanderHHansen — CHANGES_REQUESTED** [review]
nogle typer mangler + lidt ekstra,
vi skal også lige blive enige om, om vi undlader features fra vejman

**Review by frejaz — CHANGES_REQUESTED** [review]
Fix de par docstrings der som Alexander nævner

**Review by frejaz — APPROVED** [review]
:)

File: pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    ) -> pd.DataFrame:
+        """Adds the ground truth speed limit as a coloumn KODE_HAST_GENEREL_HAST to a dataframe by matching the osm_id coloumn.
+
+        Args:
+            fcd_data (pd.DataFrame): FCD data with osm_id coloumn
+            ground_truth_data (pd.DataFrame): Clean ground truth speed limit data with osm_id and KODE_HAST_GENEREL_HAST coloumns
+
+        Returns:
+            pd.DataFrame: Dataframe which has the same coloumns as fcd_data input but with KODE_HAST_GENEREL_HAST coloumn added.
+        """
```
Comment (AlexanderHHansen) [inline pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py:67]: Pt bruger Feature.HAST_GAELDENDE_HAST som vores target. I guess sikrer dig at kolonnen hedder det rigtige. Er der en KODE_HAST_GAELDENDE_HAST du kan bruge?

**Review by AlexanderHHansen — CHANGES_REQUESTED** [review]
enkelt ting

File: pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py (hunk 7 above)
Comment (CasperKjaerhus) [reply pipeline/preprocessing/ground_truth_processing/GroundTruthHandler.py:67]: Vi bruger bare gaeldende_hast så! :)

**Pull request merged**